packages/auth/src/core/auth/auth_impl.test.ts

```diff
@@ -786,4 +786,125 @@ describe('core/auth/auth_impl', () => {
       expect(auth._getRecaptchaConfig()).to.eql(cachedRecaptchaConfigOFF);
     });
   });
+
+  describe('AuthStateReady', () => {
+    let user: UserInternal;
+    let onAuthStateChangedCallback: sinon.SinonSpy;
+    let authStateChangedSpy: sinon.SinonSpy;
+
+    beforeEach(async () => {
+      user = testUser(auth, 'uid');
+
+      onAuthStateChangedCallback = sinon.spy();
+      auth.onAuthStateChanged(onAuthStateChangedCallback);
```

Comment:
The primary purpose of "onAuthStateChangedCallback" is to ensure that updateCurrentUser calls the callback registered by onAuthStateChanged?

IMO, that shouldn't be tested in the "AuthStateReady" test block. In this test block, we want to make sure that when authStateReady is called and we already updated a current user.. it should return right away without invoking "onAuthStateChanged".

TLDR, i think we should remove lines 798, 799 and related checks throughout the file.